src/local/butler/lint.py

```diff
@@ -15,10 +15,138 @@
 
 import os
 import sys
+import yaml
 
 from local.butler import appengine
 from local.butler import common
 
+_GOLINT_EXCEPTIONS = [
+    'types.go'  # Not all model names conform to Go naming conventions.
+]
+_LICENSE_CHECK_FILENAMES = ['Dockerfile']
+_LICENSE_CHECK_EXTENSIONS = [
+    '.bash',
+    '.c',
+    '.cc',
+    '.cpp',
+    '.css',
+    '.h',
+    '.htm',
+    '.html',
+    '.js',
+    '.go',
+    '.proto',
+    '.ps1',
+    '.py',
+    '.sh',
+    '.yaml',
+]
+_LICENSE_CHECK_IGNORE_FILENAMES = ['technology.css']
+_LICENSE_CHECK_IGNORE_DIRECTORIES = [
+    'third_party',
+    'templates',  # Generated code.
+]
+_LICENSE_CHECK_STRING = 'http://www.apache.org/licenses/LICENSE-2.0'
+_PY_TEST_SUFFIX = '_test.py'
+_PY_INIT_FILENAME = '__init__.py'
+_YAML_EXCEPTIONS = ['bad.yaml']
+
+
+def license_validate(file_path):
+  """Run license header validation."""
+  filename = os.path.basename(file_path)
+  extension = os.path.splitext(file_path)[1]
+  if (filename not in _LICENSE_CHECK_FILENAMES and
+      extension not in _LICENSE_CHECK_EXTENSIONS):
+    return
+
+  path_directories = file_path.split(os.sep)
+  for directory in _LICENSE_CHECK_IGNORE_DIRECTORIES:
+    if directory in path_directories:
+      return
+
+  source_filename = os.path.basename(file_path)
+  for check_filename in _LICENSE_CHECK_IGNORE_FILENAMES:
+    if check_filename == source_filename:
+      return
+
+  if _LICENSE_CHECK_STRING in open(file_path).read():
+    return
+
+  print 'Missing license header for %s.' % file_path
```

Comment:
nit: for future python 3 compatibility, let's use `from __future__ import print_function` to enforce parentheses here.